concrete/src/Cache/Adapter/ZendCacheDriver.php

```diff
@@ -27,27 +27,78 @@
 class ZendCacheDriver extends AbstractAdapter implements StorageInterface, FlushableInterface
 {
     /**
-     * @var string Name of the cache being used
+     * @var string Name of the cache being used when the localization is fully initialized
      */
-    private $cacheName;
+    private $finalCacheName;
+
+    /**
+     * @var \Concrete\Core\Application\Application
+     */
+    protected $app;
+
+    /**
+     * @var \Concrete\Core\Config\Repository\Repository
+     */
+    protected $config;
+
+    /**
+     * Is the localization system ready?
+     *
+     * @var bool
+     */
+    private $localizationReady;
 
     /**
      * @var int Number of seconds to consider the cache fresh before it expires
      */
     protected $cacheLifetime;
 
     /**
-     * @param string $cacheName Name of the cache being used. Defaults to cache.
-     * @param int $cacheLifetime Number of seconds to consider the cache fresh before it expires.
+     * @param string $finalCacheName Name of the cache being used when the localization is fully initialized. Defaults to cache
+     * @param int $cacheLifetime Number of seconds to consider the cache fresh before it expires
      */
-    public function __construct($cacheName = 'cache', $cacheLifetime = null)
+    public function __construct($finalCacheName = 'cache', $cacheLifetime = null)
     {
         parent::__construct();
-
-        $this->cacheName = $cacheName;
+        $this->setApplication(Application::getFacadeApplication());
```

Comment:
If we use `$app->make` or `$app->build` to create these classes, you can just implement ApplicationAwareInterface and avoid being coupled to the application facade